cli/command/plugin/install.go

```diff
@@ -54,26 +54,14 @@ func newInstallCommand(dockerCli command.Cli) *cobra.Command {
 	return cmd
 }
 
-type pluginRegistryService struct {
-	registry.Service
-}
-
-func (s pluginRegistryService) ResolveRepository(name reference.Named) (*registry.RepositoryInfo, error) {
-	repoInfo, err := s.Service.ResolveRepository(name)
+func resolvePlugnRepository(name reference.Named) (*registry.RepositoryInfo, error) {
+	repoInfo, err := registry.ParseRepositoryInfo(name)
```

Comment:
For reference; this is what `ParseRepositoryInfo` does; https://github.com/docker/cli/blob/14482589df194a86b2ee07df643ba3277b40df7d/vendor/github.com/docker/docker/registry/config.go#L413-L432

It uses an `emptyServiceConfig`, which is needed to mark `localhost` as "insecure", and `docker.io` as "official"
https://github.com/docker/cli/blob/14482589df194a86b2ee07df643ba3277b40df7d/vendor/github.com/docker/docker/registry/config.go#L57